nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/jdbc/SQLInfo.java

```diff
@@ -1004,12 +1017,35 @@ protected void postProcessSelectChildrenIdsAndTypes() {
                     + getSoftDeleteClause(tableName);
             select.setWhere(where);
             selectChildrenIdsAndTypesSql = select.getStatement();
+
+            // Build the exclude from copy part of the where clause
+            SchemaManager schemaManager = Framework.getService(SchemaManager.class);
+            String excludedTypes = schemaManager.getSpecialDocumentTypes()
+                                                .stream()
+                                                .collect(Collectors.joining("', '"));
+            if (!excludedTypes.isEmpty()) {
+                String excludeSpecialChildrenClause = " AND "
+                        + table.getColumn(Model.MAIN_PRIMARY_TYPE_KEY).getQuotedName() + " NOT IN ('" + excludedTypes
+                        + "')";
+                select.setWhere(where + excludeSpecialChildrenClause);
+            }
+            selectChildrenIdsAndTypesSqlExcludeSpecialChildren = select.getStatement();
+
             selectChildrenIdsAndTypesWhatColumns = whatColumns;
             // now only complex properties
-            where += " AND " + table.getColumn(Model.HIER_CHILD_ISPROPERTY_KEY).getQuotedName() + " = "
+            String isComplexClause = table.getColumn(Model.HIER_CHILD_ISPROPERTY_KEY).getQuotedName() + " = "
                     + dialect.toBooleanValueString(true);
-            select.setWhere(where);
+            String whereAnd = where + " AND " + isComplexClause;
+            select.setWhere(whereAnd);
             selectComplexChildrenIdsAndTypesSql = select.getStatement();
+            // now only complex properties include special children
+            if (!excludedTypes.isEmpty()) {
+                String whereAndOr = where + " AND ( " + isComplexClause + " OR "
+                        + table.getColumn(Model.MAIN_PRIMARY_TYPE_KEY).getQuotedName() + " IN ('"
+                        + String.join("', '", excludedTypes) + "') )";
```

Comment:
`excludedTypes` is already a single string with everything joined, it's wrong to re-join again (even though it works in this case)

Reply:
My bad, updated it in one place, forgot the other